### PR TITLE
staging-next: 21.11 iteration 1

### PR DESCRIPTION
**Pull request by vcunat**
Another staging-next iteration. Documentation here: https://nixos.org/manual/nixpkgs/stable/#submitting-changes-staging-next-branch

https://hydra.nixos.org/job/nixpkgs/staging-next/unstable#tabs-constituents

*Per the [release schedule](https://discourse.nixos.org/t/lets-have-a-great-21-11-release-cycle/15262/3) we are still restricted to non-breaking changes to release critical packages, and the plan is to merge in about a week (Nov 12).*

Next staging-next: #146259

### CODE REVIEW

File: pkgs/applications/misc/khard/default.nix
Comment (bobby285271) [inline pkgs/applications/misc/khard/default.nix:22]: https://github.com/NixOS/nixpkgs/pull/144407#issuecomment-959834865

```suggestion
    ruamel-base
```

Comment (SuperSandro2000) [reply pkgs/applications/misc/khard/default.nix:22]: Fixing that right now. This can be entirely removed because ruamel-yaml propgogtes it.

**Comment by Izorkin** [comment]
Curl not updating to version 7.79.1?

**Comment by happysalada** [comment]
> Curl not updating to version 7.79.1?

If you look at the diff, the update is included. However, the github interface for this PR does not show the commit. I don't understand why.

**Comment by mweinelt** [comment]
106e944e19d6fdbb0fc3d64495403b788835e1d4

image

**Comment by vcunat** [comment]
The GitHub's UI just isn't meant to handle PRs with hundreds of commits 🤷🏽

**Comment by happysalada** [comment]
Just switched my darwin system to this branch and couldn't find any regressions so far. Of course it doesn't prove anything, but it's a good sign I think.

**Comment by fabianhjr** [comment]
Ditto for linux-x86_64 since yesterday but required one version bump to build and boot: #144823.

**Comment by happysalada** [comment]
Thank you for updating zellij! I thought we had to wait until this was merged in master before doing small updates. If that is alright, there is another rust package that needs rust 1.56 to compile. If it's not okay, feel free to let me know, I'll wait.

**Comment by fabianhjr** [comment]
Hi @happysalada, I think it is ok to do so. Check the discussion around this topic on #137501

**Review by jonringer — APPROVED** [review]
From the linux side, this has 0 x86_64-linux build regressions:

```
$ nix-env --option system x86_64-linux -f /home/jon/.cache/nixpkgs-review/pr-144730-1/nixpkgs -qaP --xml --out-path --show-trace --meta
10 packages added:
icu69 (init at 69.1) lldb_12 (init at 12.0.1) lvm2-2_02 (init at 2.02.187) protobuf3_18 (init at 3.18.1) python38Packages.ruamel_base (init at 1.0.0) python38Packages.ruamel_yaml (init at 0.17.16) python38Packages.ruamel_yaml_clib (init at 0.2.4) python39Packages.ruamel_base (init at 1.0.0) python39Packages.ruamel_yaml (init at 0.17.16) python39Packages.ruamel_yaml_clib (init at 0.2.4)

35607 packages updated:
...
```

**Comment by oxalica** [comment]
I'm on 25b9e19da39ab9fd81452d1d5d0042acedd24567, `firefox-unwrapped` failed to build
Log: https://gist.github.com/oxalica/e9e563b0ee44b68b86185536f93c4a1d

**Comment by fabianhjr** [comment]
Hi @oxalica, it has been reported on and there is some ongoing discussion on: https://github.com/NixOS/nixpkgs/issues/144670

**Comment by nrdxp** [comment]
Looks like we are a day behind on this. Anything holding up the merge? I'd like to help get this merged by EOD at least, if possible.

**Comment by fabianhjr** [comment]
@nrdxp, not sure if a blocker but the only issue I am aware of is firefox: #144670.

Jon Ringer reported no regressions otherwise.

**Comment by fabianhjr** [comment]
Regarding that issue, it seems like there is some approval for a fix: https://github.com/NixOS/nixpkgs/pull/145459 (3 hours ago) that should ideally be merged into staging-next.

**Comment by sternenseemann** [comment]
> @nrdxp, not sure if a blocker but the only issue I am aware of is firefox: #144670.

There are other big ones like libreoffice and thunderbird, likely more. As it stands we absolutely *can't* merge as the `unstable` jobset doesn't succeed.

Seems like Jon's build report doesn't really hold anymore.

**Comment by vcunat** [comment]
Hydra's take on this, though some new failures were apparently temporary and will disappear on restart: https://hydra.nixos.org/eval/1721869?compare=1721771

I noticed the libreoffice regression mentioned a bit above, too.  Nothing else stood out to me, though in the comparison there still are over 10k unfinished builds.

**Comment by jonringer** [comment]
> > @nrdxp, not sure if a blocker but the only issue I am aware of is firefox: #144670.
> 
> There are other big ones like libreoffice and thunderbird, likely more. As it stands we absolutely _can't_ merge as the `unstable` jobset doesn't succeed.
> 
> Seems like Jon's build report doesn't really hold anymore.

It was my third time attempting to review it. if there are flakey  builds then it's possible i just got lucky a few times.

**Comment by sternenseemann** [comment]
@ofborg eval

**Comment by sternenseemann** [comment]
Merged again to include evaluation fix bdaf941e39921dc7ef0da9c8263c45943c850eec for the issue introduced in 6fe99723d3c23d64942e4de893ce5053198686bd while resolving a conflict in the first merge.

**Comment by yu-re-ka** [comment]
Runs on my systems just fine now. Firefox and Thunderbird work as well.

`hyperscan`, which is optionally used by `rspamd`, doesn't build.

**Comment by vcunat** [comment]
https://hydra.nixos.org/eval/1722344?compare=1722301#tabs-now-fail

We're certainly not perfect, but to me this PR seems OK for `master`.

**Pull request merged**
